### PR TITLE
Add documentation for custom headers.

### PR DESCRIPTION
**Pull request by roidelapluie**
If this PR is merged, I will add it to all the http configs on the page.

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -436,7 +447,7 @@ metric_relabel_configs:
 [ sample_limit: <int> | default = 0 ]
 
 # Per-scrape limit on number of labels that will be accepted for a sample. If
-# more than this number of labels are present post metric-relabeling, the
+# more than this number of labels are present post metric-relabeling ], the
```
Comment (bboreham) [inline docs/configuration/configuration.md:450]: Is this change intentional?  I couldn't see the opening bracket.

**Review by machine424 — APPROVED** [review]
Thanks!
(Not to block this PR but It'd be great to have tests for this, those would be used as a "proof" to users who may have questions/issues with headers. But also to catch any eventual regression on common side.)

> If this PR is merged, I will add it to all the http configs on the page.

Why not adding them in this PR?
